base/channels.jl

```diff
@@ -553,6 +704,135 @@ function take_unbuffered(c::Channel{T}) where T
     end
 end
 
+"""
+    take!(c::Channel, n::Integer[, buffer::AbstractVector])
+
+Take at most `n` items from the [`Channel`](@ref) `c` and return them in a `Vector`.
+If `buffer` is provided, it will be used to store the result. If the channel closes before
+`n` items are taken, the result will contain only the items that were available. For
+buffered channels, this operation requires fewer `lock` operations than individual `take!(c)`s.
+
+# Examples
+
+```jldoctest
+julia> c = Channel(3);
+
+julia> append!(c, 1:3);
+
+julia> take!(c, 3, zeros(Int, 3))
+3-element Vector{Int64}:
+ 1
+ 2
+ 3
+```
+
+!!! compat "Julia 1.12"
+    Requires at least Julia 1.12.
+"""
+function take!(c::Channel{T}, n::Integer, buffer::AbstractVector = Vector{T}()) where {T}
+    # for small-ish n, we can avoid resizing the buffer as we push elements
+    if n < 2^16
```

Comment:
This is fairly arbitrary. I want to resize so that we don't have a bunch of resizes later when the user is obviously wanting a specific n. But when they pass a huge `n`, it probably means they just want "all the items until the channel closes."